packages/core/src/endowmentsToolkit.js

```diff
@@ -126,14 +141,49 @@ function endowmentsToolkit({
     )
     // sort by length to optimize further steps
     whitelistedReads.sort((a, b) => a.length - b.length)
-    return makeMinimalViewOfRef(
+    const endowments = makeMinimalViewOfRef(
       sourceRef,
       whitelistedReads,
       unwrapTo,
       unwrapFrom,
       explicitlyBanned,
       allowedWriteFields
     )
+
+    if (allowRedefine.length > 0 && unwrapFrom) {
+      const expectedTarget = unwrapFrom || endowments
+      const originalDefineProperty = unwrapFrom.Object.defineProperty
```

Comment:
the test failures here indicate something is wrong

Reply:
TS is surprised we're getting Object here. The only way to avoid that is to have the whole thing be in a separate function called on the compartment globalThis with the list of fields allowed to be redefined.
This is the thing I didn't figure out here yet. It works and is guaranteed to work whenever unwrapFrom is a global reference of a compartment (which is always) but types are gonna complain.